app/src/main/java/in/ac/bits_pilani/goa/ard/activities/MainActivity.java

```diff
@@ -77,6 +125,41 @@ public void onClick(final View view) {
         toggle.syncState();
 
         navigationView.setNavigationItemSelectedListener(this);
+        navDrawerDBRef = FirebaseDatabase.getInstance().getReference(AHC.FDR_NAV_DRAWER);
+
+        final View headerView = navigationView.getHeaderView(0);
+        final ImageView navDrawerImage = ButterKnife.findById(headerView, R.id.nav_drawer_image);
+        final TextView navDrawerTitle = ButterKnife.findById(headerView, R.id.nav_drawer_title);
+        final TextView navDrawerSubtitle = ButterKnife.findById(headerView, R.id.nav_drawer_subtitle);
+
+        if (navDrawerTitleText != null) {
+            navDrawerTitle.setText(navDrawerTitleText);
+        }
+
+        if (navDrawerSubtitleText != null) {
+            navDrawerSubtitle.setText(navDrawerSubtitleText);
+        }
+
+        if (navDrawerImageURL != null) {
+            final RequestOptions navDrawerImageOptions = new RequestOptions()
+                    .placeholder(this.getDrawable(R.drawable.nav_drawer_default_image));
+
+            Glide.with(this)
+                    .load(navDrawerImageURL)
+                    .transition(DrawableTransitionOptions.withCrossFade()
+                            .crossFade(NAV_DRAWER_BACKGROUND_ANIM_DUR)
+                    )
+                    .apply(navDrawerImageOptions)
+                    .into(navDrawerImage);
+        }
+
+        navDrawerListener = new NavigationDrawerListener(
+                this,
+                navDrawerTitle,
+                navDrawerSubtitle,
+                navDrawerImage,
+                TAG);
+        navDrawerDBRef.addValueEventListener(navDrawerListener);
```

Comment:
If listener is removed in onStop, I would suggest that you add the listener in onStart